statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,538 @@
+"""
+Methods for creating summary statistics and their SE for survey data.
+
+The main classes are:
+
+  * SurveyDesign : Parent class that creates attributes for easy
+  implementation of other methods. Attributes include relabeled
+  clusters, number of clusters per strata, etc.
+
+  * SurveyStat : implements methods to calculate the standard
+  error of each statistic via either the bootstrap or jackknife
+
+  * SurveyMean : Calculates the mean of each column
+
+  * SurveyTotal : Calculates the total of each column
+
+  * SurveyQuantile: Calculates the specified quantile[s] of each column
+"""
+
+import numpy as np
+# import pandas as pd
+
+
+class SurveyDesign(object):
+    """
+    Description of a survey design, used by most methods
+    implemented in this module.
+
+    Parameters
+    -------
+    strata : array-like or None
+        Strata for each observation. If none, an array
+        of ones is constructed
+    cluster : array-like or None
+        Cluster for each observation. If none, an array
+        of ones is constructed
+    weights : array-like or None
+        The weight for each observation. If none, an array
+        of ones is constructed
+    nest : boolean
+        allows user to specify if PSU's with the same
+        PSU number in different strata are treated as distinct PSUs.
+
+    Attributes
+    ----------
+    weights : (n, ) array
+        The weight for each observation
+    nstrat : integer
+        The number of district strata
+    sclust : (n, ) array
+        The relabeled cluster array from 0, 1, ..
+    strat : (n, ) array
+        The related strata array from 0, 1, ...
+    ncs : (self.nstrat, ) array
+        Holds the number of clusters in each stratum
+    sfclust : ndarray
+        The stratum for each cluster
+    nclust : integer
+        The total number of clusters across strata
+    """
+
+    def __init__(self, strata=None, cluster=None, weights=None, rep_weights=None, nest=True):
```

Comment:
So I think the `SurveyDesign` class should have a method called `get_rep_weights(self, k)` which gets the k^th set of replicate weights. k is an integer from 0 to num_psu-1.  If `rep_weights` is provided to `SurveyDesign` as an argument, you just return its kth element.  Otherwise you would compute and return the k^th jackknife or bootstrap replicate weight vector.  This means that the `SurveyDesign` class needs to know whether it is doing bootstrap or jackknife.